core/meta/src/TClass.cxx

```diff
@@ -683,7 +683,11 @@ void TDumpMembers::Inspect(TClass *cl, const char *pname, const char *mname, con
       i = strlen(&line[0]); line[i] = ' ';
       Int_t lentit = strlen(memberTitle);
       if (lentit > 250-ktitle) lentit = 250-ktitle;
-      strncpy(&line[ktitle],memberTitle,lentit);
+      assert(sizeof(line) > 250);
+      assert(sizeof(line) > (ktitle+lentit+1));
+      // kline (sizeof(line)) is greater than 250, so we
+      // are guaranteed that this wont write out-of-bounds.
+      strlcpy(&line[ktitle],memberTitle,lentit+1);
```

Comment:
Why not use `strlcat(line, memberTitle, sizeof(line))`? Note that `strlcat()` guarantees to null terminate the string, and it takes the size of the buffer, not the size of the string to be copied. From the manpage:
```
    The strlcpy() and strlcat() functions copy and concatenate strings respectively.  They are designed to
     be safer, more consistent, and less error prone replacements for strncpy(3) and strncat(3).  Unlike
     those functions, strlcpy() and strlcat() take the full size of the buffer (not just the length) and
     guarantee to NUL-terminate the result
```

Reply:
because we want to limit to 250 - ktitle characters not to the size of the buffer ... 
Nonetheless you have a good point, now that the code is using a limiting strcpy, we can have the chopping done there.

Reply:
Another case where staying with `strncpy()` or `strncat()` is probably the better solution.

Reply:
but not applicable because of the weak gcc8 warning detection engine (or my lack of creativity to make it quiet)